includes/Services/AI_Service.php

```diff
@@ -0,0 +1,311 @@
+
+	/**
+	 * Default temperature for text generation.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var float
+	 */
+	public const DEFAULT_TEMPERATURE = 0.7;
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var \WordPress\AI\Services\AI_Service|null
+	 */
+	private static ?self $instance = null;
+
+	/**
+	 * Whether the service has been initialized.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var bool
+	 */
+	private bool $initialized = false;
+
+	/**
+	 * Cached preferred models.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var list<array{string, string}>|null
+	 */
+	private ?array $cached_models = null;
+
+	/**
+	 * Gets the singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return \WordPress\AI\Services\AI_Service The singleton instance.
+	 */
+	public static function get_instance(): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Private constructor to enforce singleton pattern.
+	 *
+	 * @since 0.1.0
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the AI service.
+	 *
+	 * This method should be called after AI_Client::init() on the WordPress 'init' hook.
+	 *
+	 * @since 0.1.0
+	 */
+	public function init(): void {
+		if ( $this->initialized ) {
+			return;
+		}
+
+		$this->initialized = true;
+
+		/**
+		 * Fires when the AI service is initialized.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param \WordPress\AI\Services\AI_Service $service The AI service instance.
+		 */
+		do_action( 'ai_service_initialized', $this );
+	}
+
+	/**
+	 * Checks if an AI provider is available and configured.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return bool True if a provider is available, false otherwise.
+	 */
+	public function is_available(): bool {
+		/**
+		 * Filters whether the AI service is available.
+		 *
+		 * Allows developers or hosts to override availability detection,
+		 * for example when using pre-configured providers.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param bool|null $available Null to use default detection, or boolean to override.
+		 */
+		$available = apply_filters( 'ai_service_available', null );
+
+		if ( null !== $available ) {
+			return (bool) $available;
+		}
+
+		// Check if any provider credentials are configured.
+		$credentials = get_option( 'wp_ai_client_provider_credentials', array() );
+
+		return ! empty( $credentials );
+	}
+
+	/**
+	 * Generates text from a prompt.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param string               $prompt  The prompt to generate text from.
+	 * @param array<string, mixed> $options Optional. Generation options. {
+	 *     @type string $system_instruction System instruction for the AI.
+	 *     @type float  $temperature        Temperature for generation (0.0-2.0).
+	 *     @type int    $max_tokens         Maximum tokens to generate.
+	 * }
+	 * @return string|\WP_Error The generated text or WP_Error on failure.
+	 */
+	public function generate_text( string $prompt, array $options = array() ) {
```

Comment:
I don't see the benefit of this method (and `generate_texts` similarly). Why are we not leveraging the APIs that the WordPress AI Client SDK gives us?

I would suggest for the center piece of this class to be the `create_prompt` method that sets up a prompt builder with all of the defaults applied. And allow passing an optional array of config options to that method.

What the consuming code then does with that prompt builder is up to them; they can use all the underlying APIs as needed.

Something like:
```php
$prompt_builder = $ai_service->create_prompt( $prompt, $options );
$text = $prompt_builder->generate_text();
```

Or, make it a single statement:
```php
$text = $ai_service->create_prompt( $prompt, $options )->generate_text();
```